src/include/sbpl_cpr/utils/list.h

```diff
@@ -685,4 +685,220 @@ class CSlidingBucket
     }
 };
 
+// sliding bucket for templated data

```

Comment:
This class follows the conventions of CSlidingBucket but will hold a templated data type rather than a fixed object.